retrofit-adapters/rxjava/src/main/java/retrofit/ObservableExecutionAdapter.java

```diff
@@ -0,0 +1,107 @@
+package retrofit;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import rx.Observable;
+import rx.Subscriber;
+import rx.functions.Action0;
+import rx.functions.Func1;
+import rx.subscriptions.Subscriptions;
+
+/**
+ * TODO docs
+ */
+public final class ObservableExecutionAdapter implements CallAdapter {
+  /**
+   * TODO
+   */
+  private static ObservableExecutionAdapter create() {
+    return new ObservableExecutionAdapter();
+  }
+
+  private ObservableExecutionAdapter() {
+  }
+
+  @Override public Type parseMessageType(Type returnType) {
+    if (Types.getRawType(returnType) != Observable.class) {
+      return null;
+    }
+    if (returnType instanceof ParameterizedType) {
+      return Types.getParameterUpperBound((ParameterizedType) returnType);
+    }
+    throw new IllegalStateException("Observable return type must be parameterized"
+        + " as Observable<Foo> or Observable<? extends Foo>");
+  }
+
+  @Override public <T> Observable<?> adapt(final Call<T> call, final Packaging packaging) {
+    Observable<Response<T>> responseObservable = Observable.create(new CallOnSubcribe<T>(call));
+    switch (packaging) {
+      case RESPONSE:
+        return responseObservable;
+
+      case RESULT:
+        return responseObservable //
+            .map(new Func1<Response<T>, Result<T>>() {
+              @Override public Result<T> call(Response<T> response) {
+                return Result.fromResponse(response);
+              }
+            })
+            .onErrorReturn(new Func1<Throwable, Result<T>>() {
+              @Override public Result<T> call(Throwable throwable) {
+                return Result.fromError(throwable);
+              }
+            });
+
+      case NONE:
+        return responseObservable //
+            .flatMap(new Func1<Response<T>, Observable<T>>() {
```

Comment:
[`map`][1] converts each `Foo` to a `Bar`. [`flatMap`][2] converts each `Foo` to an `Observable<Bar>` and "flattens" each back into a single stream.

Normally in this case I would use `map`, but I have no way of throwing the checked `IOException` for it to show up as an `onError`. I leverage `flatMap` to either return a stream that has a single "`Bar`" value which will be flattened and thus behave like a `map` OR I return a stream that immediately calls `onError`.

 [1]: http://reactivex.io/documentation/operators/map.html
 [2]: http://reactivex.io/documentation/operators/flatmap.html